### PR TITLE
Feature/update profiles dispatch in firebase after sign up

### PR DESCRIPTION
**Pull request by yanvow**

### CODE REVIEW

File: app/src/main/java/com/github/factotum_sdp/factotum/data/LoginDataSource.kt [hunk 1]
```diff
@@ -39,27 +37,27 @@ class LoginDataSource {
         dbRef.child(DISPATCH_DB_PATH).get().addOnSuccessListener {
             if (!it.exists()) {
                 profilesResultFuture.complete(
-                    Result.Error(IOException("Error retrieving profiles"))
+                    Result.Error(IOException("Error retrieving users"))
                 )
                 return@addOnSuccessListener
             }
-            val profileList = it.value as List<*>
+            val profileList = (it.value as Map<*, *>).values.toList()
```
Comment (TimVW7) [inline app/src/main/java/com/github/factotum_sdp/factotum/data/LoginDataSource.kt:44]: My compiler doesn't seem to accept this line, should be investigated further, but if it works, on the ci I guess it shouldn't be a problem. It says that it.value being an array list cannot be cast to a map.

**Review by TimVW7 — CHANGES_REQUESTED** [review]
Very nice cleaning of the code, you should change the mainactivitytest import for some of the tests to pass. Otherwise, once everything passes on the ci should be good to go.

File: app/src/main/java/com/github/factotum_sdp/factotum/data/LoginDataSource.kt (hunk 1 above)
Comment (yanvow) [reply app/src/main/java/com/github/factotum_sdp/factotum/data/LoginDataSource.kt:44]: I'm changing the way I'm storing profiles in the db.

**Review by TimVW7 — APPROVED** [review]
Great, ready for merge !

**Pull request merged**